src/Hazelcast.Net/Serialization/SerializationService.cs

```diff
@@ -174,6 +179,19 @@ internal sealed partial class SerializationService : IDisposable, IReadObjectsFr
 
         public Endianness Endianness { get; }
 
+        private void EnsureConstantNotOverridenWithCompact()
+        {
+            foreach (var error in _constantByType.Keys
+                         .Select(x => (oops: _compactSerializer.TryGetSerializer(x, out var y), type: x, serializer: y))
+                         .Where(x => x.oops))
+            {
+                var serializer = error.serializer == null ? "the reflection serializer" : $"serializer {error.serializer}";
+                throw new ConfigurationException($"Cannot register {serializer} for type {error.type} " +
+                                                 "because this type is a constant type and SerializationOptions.AllowOverrideDefaultSerializers " +
```

Comment:
Where do you check `AllowOverrideDefaultSerializers` ?

Reply:
Hold on, good catch, this is an error in the message - we don't have `AllowOverrideDefaultSerializers` in .NET I believe, Metin said don't put it, it's very Java-specific. Going to fix the message.

Reply:
have pushed an update that fixes this